Python/frame.c

```diff
@@ -16,11 +16,17 @@ _PyFrame_Traverse(_PyInterpreterFrame *frame, visitproc visit, void *arg)
     Py_VISIT(frame->f_funcobj);
     Py_VISIT(_PyFrame_GetCode(frame));
    /* locals */
-    PyObject **locals = _PyFrame_GetLocalsArray(frame);
+    _PyStackRef *locals = _PyFrame_GetLocalsArray(frame);
     int i = 0;
     /* locals and stack */
     for (; i <frame->stacktop; i++) {
-        Py_VISIT(locals[i]);
+#ifdef Py_GIL_DISABLED
+        if ((locals[i].bits & Py_TAG_DEFERRED) &&
```

Comment:
```suggestion
        assert((locals[i].bits & Py_TAG_DEFERRED) == 0);
```

If the locals are being traversed, that means the locals are embedded in a heap object, so they should not be deferred.

Reply:
Generators are heap objects and can contain locals that are deferred.